pandas/core/indexes/base.py

```diff
@@ -2622,41 +2622,42 @@ def _union(self, other, sort):
         lvals = self._values
         rvals = other._values
 
-        if sort is None and self.is_monotonic and other.is_monotonic:
-            try:
-                result = self._outer_indexer(lvals, rvals)[0]
-            except TypeError:
-                # incomparable objects
-                result = list(lvals)
-
-                # worth making this faster? a very unusual case
-                value_set = set(lvals)
-                result.extend([x for x in rvals if x not in value_set])
-                result = Index(result)._values  # do type inference here
-        else:
-            # find indexes of things in "other" that are not in "self"
-            if self.is_unique:
-                indexer = self.get_indexer(other)
-                indexer = (indexer == -1).nonzero()[0]
-            else:
-                indexer = algos.unique1d(self.get_indexer_non_unique(other)[1])
-
-            if len(indexer) > 0:
-                other_diff = algos.take_nd(rvals, indexer, allow_fill=False)
-                result = concat_compat((lvals, other_diff))
-
+        # if sort is None and self.is_monotonic and other.is_monotonic:
+        try:
```

Comment:
is there a viable way to decrease the amount of code inside the "try"?  its nice to have it be clear what the failure modes we expect to raise

Reply:
Hm, we could define a helper function which executes the except block and use two try blocks

Alternatively we could execute the new code on both cases, would not have any effect but would cost performance

Reply:
can you identify parts of this that you specifically want to catch in the try/except and move stuff after that into the "else" if a try/except/else?

Reply:
When we duplicate the if condition we can move the resorting into the else block. Done in the newest commit